Kernel/Arch/i686/Bootloader/stivale2.c

```diff
@@ -167,20 +168,20 @@ static void stage1MapBootloader(void)
     struct DirectoryEntry* bootDirectoryEntry = &pageDirectory.entries[bootDirectoryEntryIdx];
     bootDirectoryEntry->present = 1;
     bootDirectoryEntry->readWrite = 1;
-    bootDirectoryEntry->tableAddr = (uint32_t)&bootPageTable;
+    bootDirectoryEntry->tableAddr = (uint32_t)&bootPageTable >> PAGE_TABLE_ENTRY_SHIFT;
 
     // Get the length of the bootloader information
     struct stivale2_tag *tags = (struct stivale2_tag *)(uintptr_t)stivale2Info->tags;
     uint32_t stivale2InfoLength = stivale2_mmap_helper(tags);
     uint32_t stivale2InfoEnd = stivale2InfoAddr + stivale2InfoLength;
 
     // Map in the entire bootloader information linked list
-    size_t bootMemoryIdx = 0;
     for (uintptr_t addr = stivale2InfoAddr; addr < stivale2InfoEnd; addr += ARCH_PAGE_SIZE) {
-        struct TableEntry* bootTableEntry = &bootPageTable.pages[bootMemoryIdx++];
+        size_t bootMemoryIdx = addr >> PAGE_TABLE_ENTRY_SHIFT;
+        struct TableEntry* bootTableEntry = &bootPageTable.pages[bootMemoryIdx & 0xfff];
```

Comment:
I feel like maybe we should explain why we're doing `& 0xFFF` here (because it isn't immediately obvious to me)